src/make_trade.ts

```diff
@@ -277,7 +277,10 @@ async function giveAllowanceIfNecessary(
   const erc20 = await toERC20(sellToken.address, ethers);
   const allowance = await erc20.allowance(trader.address, allowanceManager);
   if (allowance.lt(sellAmount)) {
-    await erc20.connect(trader).approve(allowanceManager, MAX_ALLOWANCE);
+    const tx = await erc20
+      .connect(trader)
+      .approve(allowanceManager, MAX_ALLOWANCE);
+    await tx.wait;
```

Comment:
```suggestion
    await tx.wait();
```

`wait` is a function right? Its annoying that `await anyValue` works.

Reply:
https://docs.ethers.io/v5/api/providers/types/#providers-TransactionResponse

Reply:
Very good catch!

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2F3o7ZesJdQccI0sO3fy%2Fgiphy-downsized-medium.gif&token=catch" data-gitmeme-token="catch"><img src="https://media3.giphy.com/media/3o7ZesJdQccI0sO3fy/giphy-downsized-medium.gif" title="Created by gitme.me with /catch"/></a>